src/rust/engine/ui/src/console_ui.rs

```diff
@@ -78,19 +82,13 @@ impl ConsoleUI {
     }
   }
 
-  pub fn with_console_ui_disabled<F: FnOnce() -> T, T>(&self, f: F) -> T {
-    if let Some(instance) = &self.instance {
-      instance
-        .multi_progress
-        .set_draw_target(ProgressDrawTarget::hidden());
-      let res = f();
-      instance
-        .multi_progress
-        .set_draw_target(Self::default_draw_target());
-      res
-    } else {
-      f()
+  pub fn with_console_ui_disabled<F: FnOnce() -> T, T>(&mut self, f: F) -> T {
+    if self.instance.is_some() {
+      self.teardown();
     }
+    let interval = Duration::from_millis(1000 / Self::render_rate_hz());
+    std::thread::sleep(interval);
```

Comment:
It feels like rather than sleeping, `teardown` should be made synchronous: ie, that it should guarantee that the UI has been drawn for the last time before it returns.

Now that we don't join the background thread anymore, what is guaranteeing that the `MultiProgress` has been stopped?